lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -68,7 +84,28 @@ internal sealed class ContextRuntime : IDisposable
                 Optional<ContextRuntime> parentContext)
         {
             _serviceInjector = serviceInjector;
-            _injectedServices = Optional<ISet<object>>.Of(serviceInjector.GetNamedInstance<ServicesSet, ISet<object>>());
+
+            // Note that for Service objects and handlers, we are not merging them into a separate
+            // class (e.g. ServiceContainer) due to the inability to allow service stacking if an instance 
+            // of such a class were to be materialized. i.e. if a ServiceContainer object were initialized
+            // and a child ServiceConfiguration is submitted, when the child service injector tries to
+            // get the relevant handlers and services set, it will get the same set of handlers as
+            // previously instantiated by the parent injector, and thus will not allow the stacking
+            // of ServiceConfigurations.
+            _injectedServices = serviceInjector.GetNamedInstance<ServicesSet, ISet<object>>();
```

Comment:
Because these are context handlers as specified in the service configuration, not in the context configuration.